modules.d/03systemd-hostnamed/module-setup.sh

```diff
@@ -0,0 +1,46 @@
+#!/bin/bash
+
+# called by dracut
+check() {
+    [[ $mount_needs ]] && return 1
+
+    if ! dracut_module_included "systemd"; then
+        derror "systemd-hostnamed needs systemd in the initramfs"
+        return 1
+    fi
+
+    return 0
+}
+
+# called by dracut
+depends() {
+    echo "systemd systemd-networkd"
+}
+
+installkernel() {
+    return 0
+}
+
+# called by dracut
+install() {
+    local _mods
+
+    inst_multiple -o \
+       $systemdutildir/systemd-hostnamed \
+       $systemdsystemunitdir/systemd-hostnamed.service \
+       hostname hostnamectl
+
+    if [[ $hostonly ]]; then
+       inst_multiple -H -o \
+           /etc/hosts \
+           /etc/hostname \
+    	   /etc/machine-info \
+           ${NULL}
+    fi
+
```

Comment:
Shouldn't we also install:

```
/usr/share/dbus-1/system.d/org.freedesktop.hostname1.conf
/usr/share/dbus-1/system-services/org.freedesktop.hostname1.service
/usr/lib/systemd/system/dbus-org.freedesktop.hostname1.service
```

?

Reply:
@bengal Not needed afaik

Reply:
Oh, right; they are only needed if there is D-Bus support in the initrd (which is not there yet, but is being added in #900).